library/MachineControl.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from urllib.parse import quote
+
+from library.common import *
+from library.NetworkControl import NetworkControl
+from library.StateControl import StateControl
+
+class MachineControl(object):
+    def __init__(self, requester, machine_name):
+        self.requester = requester
+
+        try:
+            self.machine_id = self._get_id(machine_name)
+            self.networkcontrol = NetworkControl(self.requester, self.machine_id)
+        except Exception as err:
+            raise err
+
+        self.statecontrol = StateControl(self.requester, self.machine_id)
+
+    def _get_id(self, machine_name):
+        """ Look up machine by name """
+        q = '(= name "{}")'.format(quote(machine_name))
+        url = "/api/v1/machine?q={}&show_all=false".format(q)
+        try:
+            data = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        return data[0]['id']
+
+    def get_network_info(self, command, interface_name):
+        if command == 'getip':
+            return self.networkcontrol.get_ip()
```

Comment:
Honestly, it would have been a lot simpler if we just had "getall" as the only option, return a dictionary and then the caller can do whatever it wants with it.

This string based query is remnant of the Ansible connection and is unnecessary.

Reply:
Some users do only want the IP or MAC of a machine, should those have to sed the output ?
(e.g. ohpc_install and testsuite)

Reply:
sed? They will receive a dictionary, not text. They can do whatever they want with the dictionary. We should *not* return formatted text at all from the API.

Reply:
The CLI needs to return formatted text though !